python/cudf/benchmarks/API/bench_dataframe.py

```diff
@@ -41,10 +41,10 @@ def bench_merge(benchmark, dataframe, num_key_cols):
 @pytest.mark.parametrize(
     "values",
     [
-        range(1000),
-        {f"key{i}": range(1000) for i in range(10)},
-        cudf.DataFrame({f"key{i}": range(1000) for i in range(10)}),
-        cudf.Series(range(1000)),
+        range(50),
+        {f"{string.ascii_lowercase[i]}": range(50) for i in range(10)},
+        cudf.DataFrame({f"{string.ascii_lowercase[i]}": range(50) for i in range(10)}),
```

Comment:
This problem isn’t new in this PR, but we should try to avoid constructing objects with GPU-resident data in the `parametrize` call. Doing so causes GPU allocations while _collecting_ tests, which happens before runtime and occurs for every test regardless of whether it is selected (e.g. with `pytest -k`). An alternative solution is to pass a class and arguments, then construct the object at runtime. Even passing a parameter that is a lambda like `lambda: cudf.DataFrame(…)` and calling that in the function body will suffice to make the call lazier and execute at runtime instead of eagerly at collection time.

Reply:
That is a sufficient solution for deferring execution, yes! (Thank you for reading between the lines here, I wrote the original comment on mobile and couldn't write the actual code.)

Reply:
tl;dr I'm fine with using this suggestion for now. I don't think we should try to address the lazy evaluation question more generally in this PR because there are open ongoing discussions around it that also impact all our tests.

Avoiding collection-time creation of parametrized inputs is one of the many reasons that I advocated the use of `pytest-cases`, which provides more elegant solutions for this, but we tabled that discussion in #11122 to prioritize getting the rest of that PR merged before my vacation. I plan to pick that discussion up again soon, and depending on where we come down we will either switch to using cases or enshrine some preferred solution for this issue into our developer documentation. I think that your current solution is fine in this instance, but there are more complex cases where the results are unreadable and I would like to figure out a consistent solution to apply across all our tests/benchmarks.